homeassistant/components/enasolar/config_flow.py

```diff
@@ -0,0 +1,218 @@
+"""EnaSolar solar inverter configuration."""
+from __future__ import annotations
+
+import contextlib
+import logging
+import socket
+from typing import Any
+
+import aiohttp
+import pyenasolar
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import dt as dt_util
+
+from .const import (
+    CAPABILITY,
+    CONF_CAPABILITY,
+    CONF_DC_STRINGS,
+    CONF_HOST,
+    CONF_MAX_OUTPUT,
+    CONF_NAME,
+    CONF_SUN_DOWN,
+    CONF_SUN_UP,
+    DC_STRINGS,
+    DEFAULT_HOST,
+    DEFAULT_NAME,
+    DEFAULT_SUN_DOWN,
+    DEFAULT_SUN_UP,
+    DOMAIN,
+    MAX_OUTPUT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _get_ip(host):
+    """Get the ip address from the host name."""
+    with contextlib.suppress(socket.gaierror):
+        return socket.gethostbyname(host)
+
+
+class EnaSolarConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """User Configuration of EnaSolar Integration."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        # Only exposes methods, no significant
+        # code is executed during instantiation
+        self._enasolar = pyenasolar.EnaSolar()
+        self._data: dict[str, Any] = {}
+
+    def _conf_for_inverter_exists(self, serial) -> bool:
+        """Return True if inverter exists in configuration."""
+        return any(
+            entry
+            for entry in self._async_current_entries(include_ignore=False)
+            if serial == entry.unique_id
+        )
+
+    async def _try_connect(self, host):
+        """Needed to mock connection when running tests."""
+        error = None
+        try:
+            await self._enasolar.interogate_inverter(host)
+        except aiohttp.client_exceptions.ClientConnectorError:
+            error = "cannot_connect"
+        except aiohttp.client_exceptions.ClientResponseError:
+            error = "unexpected_response"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            error = "unknown"
+        return error
+
+    def get_serial_no(self):
+        """Needed to mock serial_no when running tests."""
+        return self._enasolar.get_serial_no()
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
```

Comment:
```suggestion
    def async_get_options_flow(config_entry) -> FlowHandler:
```

Reply:
Seems missing in many places

Reply:
Am new to python but will see if I can clean up.